### PR TITLE
Add `IInterlockMutable<'a>`

### PR DESCRIPTION
**Pull request by jeroldhaas**
This adds an interface for consuming mutables that implement interlock (with Swap), possibly for internal use in `State<'a>` and other sundry future-types.

---
This change is [Reviewable](https://reviewable.io/reviews/reedcopsey/gjallarhorn/29)

### CODE REVIEW

**Comment by jeroldhaas** [comment]
This history direly needs squashed, however I keep getting `noop` in `git rebase -I` for some reason...

**Comment by ReedCopsey** [comment]
I don't mind the extra commits... I can squash the PR.

File: src/Gjallarhorn/State.fs [hunk 1]
```diff
@@ -82,3 +83,37 @@ type State<'TModel,'TMsg> (initialState : 'TModel, update : 'TMsg -> 'TModel ->
...
+
+    interface IObservable<'a> with
+        member this.Subscribe obs = deps.Subscribe(obs,this)
+    interface ITracksDependents with
+        member this.Track dep = deps.Add (dep,this)
+        member this.Untrack dep = deps.Remove (dep,this)
+    interface IDependent with
+        member __.UpdateDirtyFlag _ = ()
+        member this.HasDependencies with get() = deps.HasDependencies
+    interface ISignal<'a>
```
Comment (ReedCopsey) [inline src/Gjallarhorn/State.fs]: appears to be missing...?

File: src/Gjallarhorn/State.fs [hunk 2]
```diff
@@ -82,3 +83,37 @@ type State<'TModel,'TMsg> (initialState : 'TModel, update : 'TMsg -> 'TModel ->
...
+        with get() = v
+        and set(value) =
+            let result = Interlocked.Exchange<'a>(&v, value)
+            deps.MarkDirty(this)
+
+    member this.Swap f =
+        swap f
+        deps.MarkDirty(this)
+        
+    override this.Finalize() =
```
Comment (ReedCopsey) [inline src/Gjallarhorn/State.fs]: should also implement IDisposable to clear deposit deterministically

**Comment by jeroldhaas** [comment]
I'll make the changes as requested here and combine both the interface and the implementation into one single PR (#28) to avoid confusion.

File: src/Gjallarhorn/State.fs (hunk 1 above)
Comment (jeroldhaas) [inline src/Gjallarhorn/State.fs]: aliasing to `this.Value`'s set, generic get

**Pull request closed without merging**